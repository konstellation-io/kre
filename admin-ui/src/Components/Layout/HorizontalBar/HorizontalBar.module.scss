@import 'Styles/colors';
@import 'Styles/variables';

.container {
  width: 100%;
  display: flex;
  flex-direction: row-reverse;
  background-color: $bg-color-dark;

  > div:not(:last-child) {
    margin-left: $grid-unit;
  }
}
