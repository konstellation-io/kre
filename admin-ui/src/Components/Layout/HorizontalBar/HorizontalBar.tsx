import React, { FunctionComponent, ReactElement } from 'react';

import cx from 'classnames';
import styles from './HorizontalBar.module.scss';

type Props = {
  children: ReactElement | ReactElement[];
  className?: string;
};

const HorizontalBar: FunctionComponent<Props> = ({ children, className }) => (
  <div className={cx(className, styles.container)}>{children}</div>
);

export default HorizontalBar;
