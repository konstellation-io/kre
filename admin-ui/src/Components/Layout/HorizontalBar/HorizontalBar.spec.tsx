import React from 'react';
import HorizontalBar from './HorizontalBar';
import { shallow } from 'enzyme';

describe('HorizontalBar', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <HorizontalBar>
        <div className="childrenA">A</div>
        <div className="childrenB">B</div>
        <div className="childrenC">C</div>
      </HorizontalBar>
    );
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('shows children components', () => {
    expect(wrapper.exists('.childrenA')).toBeTruthy();
    expect(wrapper.exists('.childrenB')).toBeTruthy();
    expect(wrapper.exists('.childrenC')).toBeTruthy();
  });
});
