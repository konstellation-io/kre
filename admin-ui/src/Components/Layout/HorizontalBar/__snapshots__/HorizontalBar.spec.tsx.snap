// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalBar matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="childrenA"
  >
    A
  </div>
  <div
    className="childrenB"
  >
    B
  </div>
  <div
    className="childrenC"
  >
    C
  </div>
</div>
`;
